File: src/main/rst/examples/Chapter4/ruby/ExplicitWaitExample01.rb
require 'rubygems' # not required for ruby 1.9 or if you installed without gem
require 'selenium-webdriver'

driver = Selenium::WebDriver.for :firefox
driver.get "http://somedomain/url_that_delays_loading"

wait = Selenium::WebDriver::Wait.new(:timeout => 10) # seconds
begin
element = wait.until { driver.find_element(:id => "some-dynamic-element") }
ensure
driver.quit
end

File: src/main/rst/examples/Chapter4/Javascript/ExplicitWaitExample01.js
var webdriver = require('selenium-webdriver');
var until = webdriver.until;

var driver = new webdriver.Builder().forBrowser('firefox').build();
driver.get('http://somedomain/url_that_delays_loading');

var element = driver.wait(until.elementLocated(webdriver.By.name('some-dynamic-element')), 10000);